tests/server/test_resource_details.py

```diff
@@ -0,0 +1,310 @@
+    assert result.result["data"]["first_generated_version"] == 2
+    generated_time = datetime.datetime.strptime(result.result["data"]["first_generated_time"], "%Y-%m-%dT%H:%M:%S.%f").replace(
+        tzinfo=datetime.timezone.utc
+    )
+    assert generated_time == cm_times[1].astimezone(datetime.timezone.utc)
+    deploy_time = datetime.datetime.strptime(result.result["data"]["last_deploy"], "%Y-%m-%dT%H:%M:%S.%f").replace(
+        tzinfo=datetime.timezone.utc
+    )
+    assert deploy_time == resources[env.id][multiple_requires][3].last_deploy.astimezone(datetime.timezone.utc)
+    assert result.result["data"]["attributes"] == resources[env.id][multiple_requires][2].attributes
+    assert result.result["data"]["requires_status"] == {
+        "std::Directory[internal,path=/tmp/dir1],v=2": "deploying",
+        "std::File[internal,path=/tmp/dir1/file2],v=2": "deployed",
+    }
+    no_requires = ids["no_requires"]
+    result = await client.resource_details(env.id, no_requires)
+    assert result.code == 200
+    assert result.result["data"]["first_generated_version"] == 2
+    generated_time = datetime.datetime.strptime(result.result["data"]["first_generated_time"], "%Y-%m-%dT%H:%M:%S.%f").replace(
+        tzinfo=datetime.timezone.utc
+    )
+    assert generated_time == cm_times[1].astimezone(datetime.timezone.utc)
+    deploy_time = datetime.datetime.strptime(result.result["data"]["last_deploy"], "%Y-%m-%dT%H:%M:%S.%f").replace(
+        tzinfo=datetime.timezone.utc
+    )
+    assert deploy_time == resources[env.id][no_requires][2].last_deploy.astimezone(datetime.timezone.utc)
+    assert result.result["data"]["attributes"] == resources[env.id][no_requires][2].attributes
+    assert result.result["data"]["requires_status"] == {}
+
+    single_requires = ids["single_requires"]
+    result = await client.resource_details(env.id, single_requires)
+    assert result.code == 200
+    assert result.result["data"]["first_generated_version"] == 3
+    generated_time = datetime.datetime.strptime(result.result["data"]["first_generated_time"], "%Y-%m-%dT%H:%M:%S.%f").replace(
+        tzinfo=datetime.timezone.utc
+    )
+    assert generated_time == cm_times[2].astimezone(datetime.timezone.utc)
+    deploy_time = datetime.datetime.strptime(result.result["data"]["last_deploy"], "%Y-%m-%dT%H:%M:%S.%f").replace(
+        tzinfo=datetime.timezone.utc
+    )
+    assert deploy_time == resources[env.id][single_requires][2].last_deploy.astimezone(datetime.timezone.utc)
+    assert result.result["data"]["attributes"] == resources[env.id][single_requires][2].attributes
+    assert result.result["data"]["requires_status"] == {"std::Directory[internal,path=/tmp/dir1],v=2": "deploying"}
+
+    result = await client.resource_details(env.id, "non_existing_id")
+    assert result.code == 404
+    unreleased_resource = ids["unreleased"]
+    result = await client.resource_details(env.id, unreleased_resource)
+    assert result.code == 404
```

Comment:
Why a 404? The resource does exist?

Reply:
I'm not sure how to handle this case: when you get a request with a valid `resource_id`, but that resource has never been released yet.
My rationale was that the endpoint only deals with released resources, so if it's not released it's the same as if it wasn't even there. I'm open to other suggestions as well.

(Note: I don't know how often, if ever, this would happen, that someone gets the id of an unreleased resource and queries the details with this endpoint)

Reply:
@bartv do you have an opinion on this?

Reply:
At this point 404 is fine for me. The resource table will never show unreleased resources and thus details will never be requested.